### PR TITLE
JS2Week1 Homework

### PR DESCRIPTION
**Pull request by CharlesKimMaina**

### CODE REVIEW

File: javascript/javascript2/week1/øfunctionApp.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+<body>
+    <script>
+       const danishWords = ["billie", "plante", "kaffe", "bog", "ø", "planetarium"];
+
+function getShortest() {
+  let shortest = danishWords.sort((a, b) => a.length - b.length)[0];
+  return shortest;
+}
+
+console.log(getShortest());
```
Comment (Rassibassi) [inline javascript/javascript2/week1/øfunctionApp.html]: this works, however, on very large arrays. a for loop would be a faster algorithm than using sort.

File: javascript/javascript2/week1/averagemedianApp.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+console.log(getAverageMedian());
+
+
+
+
+    </script>
+</body>
+</html>
```
Comment (Rassibassi) [inline javascript/javascript2/week1/averagemedianApp.html]: Good work :)

File: javascript/javascript2/week1/spiritAnimal.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+</head>
+<body>
+
+    <h1>thetet</h1>
+
+<input id="nameInput" placeholder="type your name"></input>
+<input type="submit" id="nameId">
+    <script>
+        const nameInput = document.getElementById("nameInput");
+const nameId = document.getElementById("nameId");
```
Comment (Rassibassi) [inline javascript/javascript2/week1/spiritAnimal.html]: make sure you have the indentation right :)

File: javascript/javascript2/week1/spiritAnimal.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const randomNames = ["jumbo", "horse", "lion", "rabbit"];
+
+nameId.addEventListener("click", () => {
+  let randomIndex = Math.floor(Math.random() * randomNames.length);
+  let randomName = randomNames[randomIndex];
+  let h1 = document.createElement("h1");
+  h1.appendChild(document.createTextNode(nameInput.value));
+  h1.appendChild(document.createTextNode("-"));
+  h1.appendChild(document.createTextNode(randomName));
+  body.appendChild(h1);
```
Comment (Rassibassi) [inline javascript/javascript2/week1/spiritAnimal.html]: calling 3 times `appendChild` with a `createTextNode` is a bit over the top.

`h1.innerText = `${nameInput.value} - ${randomname}`

would do the job

File: javascript/javascript2/week1/colourChanger.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+  <p>This is my body text.</p>
+    <script>
+
+let body = document.getElementById('body')
+let body = document.querySelector ("body").addEventListener('changecolor', function(event) {
```
Comment (Rassibassi) [inline javascript/javascript2/week1/colourChanger.html]: not sure, but there is no event called `changecolor`. is this just a temp file?

File: javascript/javascript2/week1/class79excercise.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    const ul = document.createElement("ul");
+    console.log(ul);
+    ul.appendChild();
+    
+   for (let i=0; i<favouriteDishes.length;i++){
+       const liTag =document.createElement("li");
+       liTag.innerHTML = favouriteDishes[i]
+       ulTag.appendChild(liTag)       
+   }
+
```
Comment (Rassibassi) [inline javascript/javascript2/week1/class79excercise.html]: Almost!
here:

```
      const favouriteDishes =["bread", "pizza", "lasagne"];
      const body = document.querySelector("body")
      console.log(favouriteDishes);
      const ul = document.createElement("ul");
      console.log(ul);
      body.appendChild(ul);

      for (let i=0; i<favouriteDishes.length;i++){
        const liTag =document.createElement("li");
        liTag.innerHTML = favouriteDishes[i]
        ul.appendChild(liTag)       
      }
```

File: javascript/javascript2/week1/hyfbay/main.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+function renderProducts(products){
+ const productList = products.map((productName)=>{
+const li = document.createElement("li");
+li.innerHTML = productName;
+ul.appendChild(li);
+  });
+}
+console.log(renderProducts(testProductsNames));
+console.log(getAvailableProducts());
```
Comment (Rassibassi) [inline javascript/javascript2/week1/hyfbay/main.js]: This is only half of the exercise! You should only use `testProductsNames` for a quick prototype. But your page should actually show the products that are returned by the `getAvailableProducts()` function!!

`let testProductsNames = getAvailableProducts();`

File: javascript/javascript2/week1/class79excercise.html (hunk 6 above)
Comment (CharlesKimMaina) [reply javascript/javascript2/week1/class79excercise.html:24]: I have changed it to these your code snippet, its working great now,.,Thanks

File: javascript/javascript2/week1/spiritAnimal.html (hunk 4 above)
Comment (CharlesKimMaina) [reply javascript/javascript2/week1/spiritAnimal.html:27]: Done, great

File: javascript/javascript2/week1/øfunctionApp.html (hunk 1 above)
Comment (CharlesKimMaina) [reply javascript/javascript2/week1/øfunctionApp.html:17]: thanks,.for large array, i will use loop , great

**Pull request merged**